### PR TITLE
Use getrawatv to confirm ATVs

### PR DESCRIPTION
**Pull request by Warchant**

### CODE REVIEW

File: pop-miners/altchain-pop-miner/src/main/kotlin/org/veriblock/miners/pop/core/ApmOperation.kt [hunk 1]
```diff
@@ -36,6 +37,14 @@ class ApmOperation(
...
     logs, reconstituting
 ) {
+    // (mem-only) should NOT be serialized
+    var requiredConfirmations: Int? = null
+    // (mem-only) should NOT be serialized
+    var currentConfirmations: Int? = null
+    // (mem-only) should NOT be serialized
+    var atvBlock: SecurityInheritingBlock? = null
+
+
```
Comment (PCasafont) [inline pop-miners/altchain-pop-miner/src/main/kotlin/org/veriblock/miners/pop/core/ApmOperation.kt:47]: These should be the last fields, as they are less important than the serialized ones

File: pop-miners/altchain-pop-miner/src/main/kotlin/org/veriblock/miners/pop/service/ApmTaskService.kt [hunk 2]
```diff
@@ -209,22 +210,51 @@ class ApmTaskService(
...
+//                logger.info(operation, "Waiting for $requiredConfirmations required confirmations on ATV: $atvId in $chainName...")
+//            } else {
+//                logger.info(operation, "ATV: $atvId in $chainName is either confirmed or expired (tip=$lastBlockHeight, endorsed height=$endorsedBlockHeight, validity window=${operation.chain.getPayoutDelay()}")
+//            }
+//            // if ATV expired, task will fail. if confirmed, will continue.
+//            val atvBlock = operation.chainMonitor.confirmAtv(operation, atvId, requiredConfirmations)
+//            logger.info(operation, "Successfully confirmed ATV: $atvId in $chainName block ${atvBlock.hash} @ ${atvBlock.height}!")
+//
+//            operation.setAtvBlockHash(atvBlock.hash)
+//        }
```
Comment (PCasafont) [inline pop-miners/altchain-pop-miner/src/main/kotlin/org/veriblock/miners/pop/service/ApmTaskService.kt]: no need to leave this commented out

File: pop-miners/altchain-pop-miner/src/main/kotlin/org/veriblock/miners/pop/service/ApmTaskService.kt [hunk 3]
```diff
@@ -234,38 +264,58 @@ class ApmTaskService(
             verifyAltchainStatus(operation.chain.name, operation.chainMonitor)
             val miningInstruction = operation.miningInstruction
                 ?: failTask("PayoutDetectionTask called without mining instruction!")
+            val atvId = operation.atvId ?: failTask("Can't get ATV id")
```
Comment (PCasafont) [inline pop-miners/altchain-pop-miner/src/main/kotlin/org/veriblock/miners/pop/service/ApmTaskService.kt:221]: ?: failTask should be in its own line (for easier visibility and breakpointing purposes)

File: veriblock-core/src/main/java/org/veriblock/core/crypto/VbkHash.kt [hunk 4]
```diff
@@ -58,7 +58,7 @@ open class VbkHash internal constructor(bytes: ByteArray) : AnyVbkHash(bytes) {
     }
 }
 
-open class PreviousBlockVbkHash internal constructor(bytes: ByteArray) : AnyVbkHash(bytes) {
+open class PreviousBlockVbkHash constructor(bytes: ByteArray) : AnyVbkHash(bytes) {
```
Comment (PCasafont) [inline veriblock-core/src/main/java/org/veriblock/core/crypto/VbkHash.kt:61]: no need to specify the `constructor` keyword if not applying any modifiers to it

File: veriblock-core/src/main/java/org/veriblock/core/crypto/VbkHash.kt [hunk 5]
```diff
@@ -77,7 +77,7 @@ open class PreviousBlockVbkHash internal constructor(bytes: ByteArray) : AnyVbkH
     }
 }
 
-open class PreviousKeystoneVbkHash internal constructor(bytes: ByteArray) : AnyVbkHash(bytes) {
+open class PreviousKeystoneVbkHash constructor(bytes: ByteArray) : AnyVbkHash(bytes) {
```
Comment (PCasafont) [inline veriblock-core/src/main/java/org/veriblock/core/crypto/VbkHash.kt:80]: no need to specify the `constructor` keyword if not applying any modifiers to it

File: altchain-sdk/src/main/kotlin/org/veriblock/sdk/alt/SecurityInheritingChain.kt [hunk 6]
```diff
@@ -70,7 +70,7 @@ interface SecurityInheritingChain {
      * Returns the transaction for the given [txId] retrieved from the SI chain daemon,
      * or null if it does not exist.
      */
-    suspend fun getTransaction(txId: String): SecurityInheritingTransaction?
+    suspend fun getTransaction(txId: String, blockHash: String?): SecurityInheritingTransaction?
```
Comment (PCasafont) [inline altchain-sdk/src/main/kotlin/org/veriblock/sdk/alt/SecurityInheritingChain.kt:73]: Better if we make blockHash default to null

**Review by PCasafont — CHANGES_REQUESTED** [review]

**Review by PCasafont — APPROVED** [review]

**Comment by Warchant** [comment]
Closes https://github.com/VeriBlock/vbk-ri-btc/issues/406

**Pull request merged**